app/models/box.rb

```diff
@@ -271,14 +271,14 @@ def create_core_box_items!
   end
 
   def send_research_solicitation_email!
-    AutoEmailHandler.new("volunteer", self, self.designed_by)
+    AutoEmailHandler.new("volunteer", self, self.designed_by) if self.designed_by.volunteer.ok_to_email?
   end
 
   def send_assembly_solicitation_email!
-    AutoEmailHandler.new("volunteer", self, self.researched_by)
+    AutoEmailHandler.new("volunteer", self, self.researched_by) if self.researched_by.volunteer.ok_to_email?
   end
 
   def send_shipping_solicitation_email!
-    AutoEmailHandler.new("volunteer", self, self.assembled_by)
+    AutoEmailHandler.new("volunteer", self, self.assembled_by) if self.assembled_by.volunteer.ok_to_email?
```

Comment:
For these (and the volunteer_email in `app/controllers/box_request_triage_controller.rb`), the `ok_to_email` check should not be performed on the volunteer who completed the task (in this case, the `assembler`). 

Instead, the check should performed on each of the Volunteers to whom the solicitation emails (of the next step) will be sent. 

Therefore, the `ok_to_email` should be moved deeper into the `AutoEmailHandler`, which finds all the Volunteers with the next-step Permission (in this case, `Shipping`).

Reply:
Looking at `AutoEmailHandler` a little more, pretty sure we could change the query in `AutoEmailHandler#get_solicitation_recipients` so that it filters to the relevant permission level, and, through a join, checks that the `User's` associated `Volunteer` is `ok_to_email`.

Are you willing to do that in `AutoEmailHandler` as part of this PR (instead of where you'd implemented it in the `box.rb`, @sarslanoglu ?

Reply:
@maebeale Yeah for sure i can do that. I was a little busy lately, I will try to look at today

Reply:
Great, @sarslanoglu! Not trying to rush you. Just wanted to check your preferred path forward. I apologize not to have provided more detail re who we're validating it's ok_to_send and email to.